pkg/kademlia/endpoint.go

```diff
@@ -35,8 +36,14 @@ func NewEndpoint(log *zap.Logger, service *Kademlia, routingTable *RoutingTable)
 
 // Query is a node to node communication query
 func (endpoint *Endpoint) Query(ctx context.Context, req *pb.QueryRequest) (*pb.QueryResponse, error) {
+
+	duration := time.Duration(endpoint.service.queryTimeout.Seconds()) * time.Second
+
+	timedCtx, cancel := context.WithTimeout(ctx, duration)
```

Comment:
Create the timedCtx inside the if statement, we don't need it otherwise, also the conversion above is not necessary you can directly use `context.WithTimeout(ctx, endpoint.pingbackTimeout)`

Reply:
ah okay thanks